inc/shortcodes/class-googledocs.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class GoogleDocs extends Shortcode {
+
+	private static $valid_hosts = array( 'docs.google.com', 'www.google.com' );
+
+	public static function get_shortcode_ui_args() {
+		return array(
+			'label'          => esc_html__( 'Google Docs', 'shortcake-bakery' ),
+			'listItemImage'  => '<img src="' . esc_url( SHORTCAKE_BAKERY_URL_ROOT . 'assets/images/svg/icon-googledocs.svg' ) . '" />',
+			'attrs'          => array(
+				array(
+					'label'        => esc_html__( 'Document Type', 'shortcake-bakery' ),
+					'attr'         => 'type',
+					'type'         => 'select',
+					'options'      => array(
+						'document'      => esc_html__( 'Document', 'shortcake-bakery' ),
+						'spreadsheet'   => esc_html__( 'Spreadsheet', 'shortcake-bakery' ),
+						'presentation'  => esc_html__( 'Presentation', 'shortcake-bakery' ),
+						'form'          => esc_html__( 'Form', 'shortcake-bakery' ),
+						'map'           => esc_html__( 'Map', 'shortcake-bakery' ),
+					),
+					'description'  => esc_html__( 'Type of document to embed', 'shortcake-bakery' ),
```

Comment:
Good point. The only real reason for this attribute is to wire up the JS hooks, so that we don't have to display all the fields specific to presentations on all the other document types and so on. But that could really be done just as easily by reading the document type from the URL field.